Services/User/Avatar/classes/class.ilUserAvatarFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/* Copyright (c) 1998-2017 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Class ilUserAvatarFactory
+ * @author Alexander Killing <killing@leifos.de>
+ * @author Michael Jansen <mjansen@databay.de>
+ */
+class ilUserAvatarFactory
+{
+	/**
+	 * @var \ILIAS\DI\Container
+	 */
+	protected $dic;
+
+	/**
+	 * ilUserAvatarFactory constructor.
+	 * @param \ILIAS\DI\Container $dic
+	 */
+	public function __construct(\ILIAS\DI\Container $dic)
```

Comment:
Why not just inject the dependency you really require (i.e. `settings`)?

Reply:
Because maybe this PR has to be enhanced (as mentioned), e.g. I see some FileSystem requirements of the concrete user avatar classes (instead of using ilUtil), and maybe more code of ilObjUser has to be moved somehow to this new classes to handle the user pictures (if existing and published).